### PR TITLE
Add admonition to code contribution workflow to make sure working tree is clean

### PR DESCRIPTION
**Pull request by namurphy**
One issue that came up for the code contribution workflow recently is that the process of creating a new branch didn't work when there there were changes to tracked files.  This PR adds an admonition to the workflow to make sure that the working tree is clean before doing the `git checkout -b` command.

This PR also adds `sphinx-toolbox` as a Sphinx extension.  It has a lot of capabilities that can be used to make other improvements for the documentation (e.g., #2270).

### CODE REVIEW

File: docs/contributing/workflow.rst [hunk 1]
```diff
@@ -125,6 +154,8 @@ repeated multiple times.
    * ``git push`` is like sending the package off to its destination
      (i.e., GitHub).
 
+.. _create-pr
```
Comment (namurphy) [inline docs/contributing/workflow.rst:157]:
```suggestion
.. _create-pr:
```

File: docs/contributing/workflow.rst [hunk 2]
```diff
@@ -36,6 +38,33 @@ Create a new branch
...
+
+   .. code-block::
+
+      git status
+
+   If the output ends with ``nothing to commit, working tree clean``,
+   then proceed to the next step. Otherwise, click to open the next
+   section.
+
+   .. collapse:: Instructions if there are uncommitted changes
```
Comment (namurphy) [inline docs/contributing/workflow.rst:51]: The [`collapse`](https://sphinx-toolbox.readthedocs.io/en/stable/extensions/collapse.html) directive from [`sphinx-toolbox`](https://sphinx-toolbox.readthedocs.io) lets us create a collapsible section to a page which is not open by default.

**Review by ejohnson-96 — APPROVED** [review]
👍

File: docs/contributing/workflow.rst [hunk 3]
```diff
@@ -36,16 +38,43 @@ Create a new branch
...
+
+      #. :ref:`Add and commit changes <commit-changes>`,
+
+      #. Use `git stash`_ to temporarily file away the changes, or
+
+      #. Use :bash:`git reset --hard` to **permanently** remove all
+         changes to tracked files and return to the previous commit.
+
+      If there are untracked files present, then you may delete the
+      untracked files, :ref:`add and commit changes <commit-changes>,
```
[the remark this answers is not in the document]
Comment (namurphy) [reply docs/contributing/workflow.rst:65]: Great catch!

Comment (namurphy) [inline docs/contributing/workflow.rst:65]:
```suggestion
      untracked files, :ref:`add and commit changes <commit-changes>`,
```

**Review by ejohnson-96 — APPROVED** [review]
👍

**Comment by namurphy** [comment]
The new Python 3.9  & 3.10 failures were caused by NumPy and/or Astropy adopting new conventions for representing numbers, and are thus unrelated.  This is probably related to #2148.

**Pull request merged**